Code/Framework/AzToolsFramework/Tests/ComponentModeSwitcherTests.cpp

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Tests/ComponentModeTestDoubles.h>
+#include <Tests/ComponentModeTestFixture.h>
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/API/EntityCompositionRequestBus.h>
+#include <AzToolsFramework/Application/ToolsApplication.h>
+#include <AzToolsFramework/ComponentMode/ComponentModeSwitcher.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/ViewportUi/ButtonGroup.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiSwitcher.h>
+
+namespace UnitTest
+{
+    using AzToolsFramework::ComponentModeFramework::AnotherPlaceholderEditorComponent;
+    using AzToolsFramework::ComponentModeFramework::PlaceholderEditorComponent;
+    using ComponentModeSwitcher = AzToolsFramework::ComponentModeFramework::ComponentModeSwitcher;
+
+    class ComponentModeSwitcherTestFixture : public ComponentModeTestFixture
+    {
+    };
+
+    TEST_F(ComponentModeSwitcherTestFixture, AddComponentToEntityAddsComponentToSwitcher)
```

Comment:
Suggestion:  would `SelectingEntitiesAddsTheirComponentsToTheSwitcher` be a more appropriate name? As in, the adding of components to the switcher is triggered by the entity selection?


Ditto for any other test names.

Reply:
This test is more for, if an entity is already selected, and a user adds a component, does it add said component to the switcher, but I'll review the names